gobblin-core/src/main/java/gobblin/source/extractor/extract/QueryBasedSource.java

```diff
@@ -78,6 +78,7 @@
   public static final boolean DEFAULT_SOURCE_OBTAIN_TABLE_PROPS_FROM_CONFIG_STORE = false;
   private static final String QUERY_BASED_SOURCE = "query_based_source";
   public static final String WORK_UNIT_STATE_VERSION_KEY = "source.querybased.workUnitState.version";
+  public static final String IS_LAST_WORK_UNIT = "is.last.work.unit";
```

Comment:
How about "source.querybased.workUnitState.isLastWorkUnit" and IS_LAST_WORK_UNIT_KEY?